lodash.js

```diff
@@ -7496,6 +7496,29 @@
       return length ? baseSlice(array, 0, -1) : [];
     }
 
+    /**
+     * Inserts `value` at the given `index` into the `array`.
+     *
+     * @static
+     * @memberOf _
+     * @since 4.7.15
+     * @category Array
+     * @param {Array} array The array which will have an value inserted.
+     * @param {number} [index=0] The desired index of the inserted the value.
+     * @param {*} value The value to be inserted.
+     * @returns {Array} Returns a new array with the value inserted at the given index.
+     * @example
+     *
+     * _.insert(['Hey', 'Hi', 'Howdy'], 2, 'Hello')
+     * // => '['Hey', 'Hi', 'Hello', 'Howdy']'
+     * 
+     * _.insert(['Hey', 'Hi', 'Howdy'], -2, 'Hello')
+     * // => '["Hey", "Hello", "Hi", "Howdy"]'
+     */
+    function insert(array = [], index = 0, value) {
+      return [...array.slice(0, index), value, ...array.slice(index)];
```

Comment:
I think a `array.splice(index, 0, value)` in the end would be more performant although we would need to store array in a variable since it's mutating the array..